### ISSUE TITLE
Not selecting value if onChange is used

### ISSUE THREAD

**Issue by edwinlunando** [JedWatson/react-select#254]
So, my react-select component is not selecting the value. Here is my initialization code.

```
        <Select
          name="form-field-name"
          options={ areas }
          onChange={ this.areaSelected }
        />
```
This is the areaSelected.
```
  areaSelected: function(value) {
    AreaActions.setArea(value);
  },
```
The action update my state beautifully, but the select component do not updated. Is it a bug?

**Comment by brianreavis** [comment]
You need to set the `value={currentValue}` property.

**Comment by edwinlunando** [comment]
How to get the `currentValue`? I added your code and it did not work.
```
<Select
  name="form-field-name"
  options={ areas }
  onChange={ this.areaSelected }
/>
```

**Comment by brianreavis** [comment]
From the `onChange` event. In most cases, you'll want to call set `setState` with the value coming from `onChange` and then use `this.state.theValue` for the `value` property.

**Comment by edwinlunando** [comment]
I have followed your advice to change the value on `onChange`, now, it is updating the value, not the label. So, on the current selected shows the id, not the label.
```
areaSelected: function(e) {
    AreaActions.setArea(e);
    this.setState({ selectValue: e });
  },
```
and 
```
<Select
  name="form-field-name"
  options={ areas }
  onChange={ this.areaSelected }
  value={ this.state.selectValue }
/>
```
Any idea?

**Comment by tokenvolt** [comment]
Faced the same issue, it updates the value, but how do I render the label of the selected value?

**Comment by huuuda-aus** [comment]
So if I have 10 selects in a single page, will I have to have 10 handlers for that? (those selects can NOT be an array)

**Comment by jvchappell** [comment]
Hi All :)

Sorry to ask a question on a closed issue but I have got react-select working ok eventually with redux-form until I added an onChange...now despite following the advice given above (updating State and setting the value attribute) react-select still will not allow me to Select a value from the list when the onChange has been added. 

Could it be that the suggested workaround is not compatible with redux-form? If I console.log the value of  the "value" attribute it seems to get lost when I call the ReduxForm.Field component, stopping the above suggested workarounds from, well, working.

Has anyone else seen this? 

Thanks again!
